pkg/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -411,12 +412,12 @@ func (b *builder) getReadinessProbe() *corev1.Probe {
 // if the version is not set, ie.: unknown, we  consider the OneAgent to support `ReadOnlyRootFilesystem`.
 func (b *builder) isRootFsReadonly() bool {
 	if b.dk != nil &&
-		b.dk.UseReadOnlyOneAgents() &&
-		b.dk.OneAgentVersion() != "" &&
-		b.dk.OneAgentVersion() != string(status.CustomImageVersionSource) {
-		agentSemver, err := dtversion.ToSemver(b.dk.OneAgentVersion())
+		b.dk.OneAgent().IsReadOnlyOneAgentsMode() &&
+		b.dk.OneAgent().GetVersion() != "" &&
```

Comment:
I also not a big fan of `Get...` (feels like java 😅 )

but we went with that pattern because we tend to have `struct` members called `Version` and such, and then you can't create a method called `Version()`, so to avoid this we introduce this `Get` prefix to be consitant

Reply:
got it, I agree this makes sense